src/Grid.js

```diff
@@ -192,6 +206,18 @@ export default class Grid extends CollectionWrapper {
                 [main]: mainPos,
                 [cross]: crossPos
             });
+
+            const newItem = {
+                ref,
+                type: ItemWrapper,
+                componentIndex: index,
+                forceLoad: this._forceLoad,
+                ...sizes,
+                [`assigned${main.toUpperCase()}`]: mainPos,
+                [`assigned${cross.toUpperCase()}`]: crossPos,
+                [main]: tmp,
+                [cross]: tcp
+              }
             crossPos += sizes[crossDim] + (sizes[crossMarginTo] || sizes.margin || this._crossSpacing)
             this._lines[lineIndex].push(index);
             cl.push(newItem);
```

Comment:
This doesn't make sense. This method is to reposition, so the only change needed would be changing line 223:

```js
cl.push(newItem);
```

To:

```js
cl.push(item);
```

So that later on that method it can get the biggest in line. In this method it should simply patch existing items and not create new ones.